src/qml/QFieldCamera.qml

```diff
@@ -156,38 +212,176 @@ Popup {
+                currentPath = filePos === 0 ? path.substring(7) : path
+                cameraItem.state = "VideoPreview"
+              }
+            } else if (cameraItem.state == "PhotoPreview" || cameraItem.state == "VideoPreview") {
+              cameraItem.finished(currentPath)
+            }
+          }
+        }
+      }
+
+      QfToolButton {
+        id: zoomButton
+        visible: cameraItem.isCapturing
 
-      visible: cameraItem.state == "PhotoPreview" || cameraItem.state == "VideoPreview"
+        x: cameraItem.isPortraitMode ? (parent.width / 4) - (width / 2) : (parent.width - width) / 2
+        y: cameraItem.isPortraitMode ? (parent.height - height) / 2 : (parent.height / 4) * 3 - (height / 2)
 
-      anchors.right: parent.right
-      anchors.rightMargin: 5
-      anchors.verticalCenter: parent.verticalCenter
-      bgcolor: Theme.mainColor
-      round: true
+        iconColor: "white"
+        bgcolor: Qt.hsla(Theme.darkGray.hslHue, Theme.darkGray.hslSaturation, Theme.darkGray.hslLightness, 0.3)
+        round: true
+
+        text: (camera.digitalZoom * camera.opticalZoom).toFixed(1) +'X'
+        font: Theme.tinyFont
+
+        onClicked: {
+          camera.opticalZoom = 1;
+          camera.digitalZoom = 1;
+        }
+      }
+
+      QfToolButton {
+        id: flashButton
+        visible: cameraItem.isCapturing && camera.flash.supportedModes.length > 1
+
+        x: cameraItem.isPortraitMode ? (parent.width / 4) * 3 - (width / 2) : (parent.width - width) / 2
+        y: cameraItem.isPortraitMode ? (parent.height - height) / 2 : (parent.height / 4) - (height / 2)
+
+        iconSource: {
+          switch(camera.flash.mode) {
+            case Camera.FlashAuto:
+              return Theme.getThemeVectorIcon('ic_flash_auto_black_24dp');
+            case Camera.FlashOn:
+              return Theme.getThemeVectorIcon('ic_flash_on_black_24dp');
+            case Camera.FlashOff:
+              return Theme.getThemeVectorIcon('ic_flash_off_black_24dp');
+            default:
+              return'';
+          }
+        }
+        iconColor: "white"
+        bgcolor: Qt.hsla(Theme.darkGray.hslHue, Theme.darkGray.hslSaturation, Theme.darkGray.hslLightness, 0.5)
+        round: true
 
-      iconSource: Theme.getThemeIcon("ic_save_white_24dp")
+        onClicked: {
+          if (camera.flash.mode == Camera.FlashOff) {
+            camera.flash.mode = Camera.FlashOn;
+          } else {
+            camera.flash.mode = Camera.FlashOff
+          }
+        }
+      }
 
-      onClicked: cameraItem.finished(currentPath)
+      Rectangle {
+        visible: cameraItem.state == "VideoCapture" && camera.videoRecorder.recorderState != CameraRecorder.StoppedState
+
+        x: cameraItem.isPortraitMode ? captureRing.x + captureRing.width / 2 - width / 2 : captureRing.x + captureRing.width / 2 - width / 2
+        y: cameraItem.isPortraitMode ? captureRing.y - height - 20 : captureRing.y - height - 20
```

Comment:
I guess you know you are the only person in the universe understanding this `x` and `y` calculation.

Any chance that we have a "toolbar" where they are rendered sequentially? Or any kind of function calculating this?

Reply:
@suricactus , unfortunately not sir; complexity comes from the fact that we want the controls to be sitting at the bottom of the screen when in portrait and on the right edge of the screen when in landscape.